### DISCUSSION TITLE
The best place to grab table name, sql statement & sql query execution time

### DISCUSSION THREAD

**Discussion by k1r8r0wn** [jeremyevans/sequel#2257] [General]
Hello! What is the best place in code to grab the table name & sql statement to attach them into current thread of request & then inside connection pool to get query execution time?

It's just a thoughts, maybe there is more convenient option... 
Actually I need to send Yabeda metrics about number of queries & their duration aka:

table_1#select - 12331
table_1#select - 20ms

P.S. I know about logging & **caller_logging** extension, and also there is no such open telemetry extension yet.

**Comment by jeremyevans** [comment]
For timing and SQL executed, you can override Database#log_connection_yield. This won't give you the table name unless you try parsing it out of the SQL, and not all SQL statements have table names. 

To get something akin to table names, you'd need to override at the Dataset layer.  Dataset#fetch_rows is called for SELECTs, but there are other methods for other statement types.  Note that that will only provide table names if the user was using the Dataset API to execute the queries (e.g. `DB[:table_name].all` and not `DB["SELECT * FROM table_name"].all`).

Someone was trying to contribute Sequel support to opentelemetry, but it stalled about a year ago: 
https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/672

**Reply by k1r8r0wn** [reply]
After some tryings decided to patch 

> Database#log_connection_yield

 inside custom extension for metrics sending.
 
 Thanks for the suggestions, @jeremyevans, have a good one!
